client/offchain/src/lib.rs

```diff
@@ -61,65 +65,163 @@ pub trait NetworkProvider: NetworkStateInfo + NetworkPeers {}
 
 impl<T> NetworkProvider for T where T: NetworkStateInfo + NetworkPeers {}
 
+/// Special type that implements [`OffchainStorage`](offchain::OffchainStorage).
+///
+/// This type can not be constructed and should only be used when passing `None` as `offchain_db` to
+/// [`OffchainWorkerOptions`] to make the compiler happy.
+#[derive(Clone)]
+pub struct NoOffchainStorage {
+	// Ensure no one can construct this type
+	_priv: (),
+}
```

Comment:
```suggestion
pub enum NoOffchainStorage {}
```

If it's not supposed to be constructed then we can make it non-constructible. (: